packages/colorpickers/src/elements/ColorPicker/index.tsx

```diff
@@ -0,0 +1,206 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useEffect, useReducer, forwardRef, ChangeEvent, HTMLAttributes } from 'react';
+import PropTypes from 'prop-types';
+import { Label } from '@zendeskgarden/react-forms';
+import { ColorWell } from './ColorWell';
+import {
+  StyledHue,
+  StyledSliderGroup,
+  StyledInput,
+  StyledAlpha,
+  StyledSliders,
+  StyledHexField,
+  StyledRGBAField,
+  StyledInputGroup,
+  StyledPreview,
+  StyledColorPicker,
+  StyledCheckered,
+  StyledAlphaGradient,
+  StyledAlphaField,
+  StyledHueField
+} from '../../styled';
+import {
+  getInitialState,
+  reducer,
+  IRGBColor,
+  IHSVColor,
+  IColorPickerState,
+  SATURATION_CHANGE,
+  HUE_CHANGE,
+  ALPHA_SLIDER_CHANGE,
+  HEX_CHANGE,
+  RED_CHANGE,
+  GREEN_CHANGE,
+  BLUE_CHANGE,
+  ALPHA_CHANGE
+} from './reducer';
+
+export interface IColorPickerLabels {
+  hueSlider?: string;
+  alphaSlider?: string;
+  hex?: string;
+  red?: string;
+  green?: string;
+  blue?: string;
+  alpha?: string;
+}
+
+export interface IColorPickerProps
+  extends Omit<HTMLAttributes<HTMLDivElement>, 'color' | 'onChange'> {
+  /** A hex string, RGB string, RGB object, or color picker state that represents the current color */
+  color: string | IRGBColor | IColorPickerState;
+  /** A callback that is fired when an the color picker state is changed */
+  onChange?: (state: IColorPickerState) => void;
+  /** Replaces the default labels within the color picker */
+  labels?: IColorPickerLabels;
+}
+
+/**
+ * @extends HTMLAttributes<HTMLDivElement>
+ */
+export const ColorPicker = forwardRef<HTMLDivElement, IColorPickerProps>(
+  ({ color, labels = {}, onChange, ...props }, ref) => {
+    const [state, dispatch] = useReducer(reducer, getInitialState(color));
+
+    useEffect(() => {
+      onChange && onChange(state);
+    }, [state, onChange]);
```

Comment:
Is this dependency array accurate? If someone were to provide a new instance of `onChange` it would immediately get called with the current state.

Reply:
It should be accurate. If the `onChange` was called with the current state (not the updated state) then the color picker should not update. 

However, it does seem to be updating. I don't think there is a problem with this `useEffect` but happy to investigate further provided more detail.

Reply:
I think the `state` dependency is correct, I'm mostly wondering about the `onChange`. With the default `Object.is` comparison in `useEffect` changing that prop to a new function would trigger an `onChange(currentState)` to be immediately called. Is that the expected API?

Reply:
> With the default Object.is comparison in useEffect changing that prop to a new function would trigger an onChange(currentState) to be immediately called. Is that the expected API?

I think this is the expected API. By the time the `useEffect` is called, it's called with the (expected) updated state. I've also verified that the `useEffect` is called only once per state update.